libs/core/src/lib/select/select.component.ts

```diff
@@ -263,211 +308,98 @@ export class SelectComponent implements OnChanges, AfterContentInit, OnInit, OnD
 
     /** @hidden */
     writeValue(value: any): void {
-        if (this.options) {
-            this._selectValue(value, false);
-            this._changeDetectorRef.detectChanges();
-        } else {
-            // Defer the selection of the value to support forms
-            Promise.resolve().then(() => {
-                if (this.options) {
-                    this._selectValue(value, false);
-                    this._changeDetectorRef.detectChanges();
-                }
-            });
-        }
-    }
-
-    /** Returns the current trigger value if there is a selected option. Otherwise, returns the placeholder. */
-    get triggerValue(): string {
-        return this._selected ? this._selected.viewValueText : this.placeholder;
+        this.value$.next(value);
     }
 
     /** @hidden */
-    @HostListener('keydown', ['$event'])
-    keydownHandler(event: KeyboardEvent): void {
-        switch (event.code) {
-            case 'ArrowUp': {
-                event.preventDefault();
-                this._decrementFocused();
-                break;
-            }
-            case 'ArrowDown': {
-                event.preventDefault();
-                this._incrementFocused();
-                break;
-            }
-        }
-    }
-
-    /** @hidden */
-    @HostListener('window:resize')
-    resizeScrollHandler() {
-        this.calculatedMaxHeight = window.innerHeight * 0.45;
-    }
+    setSelectedOption(option: OptionComponent, controlChange: boolean) {
+        this.selected = option;
 
-    /**
-     * Selects an option by option component reference. Preferred method of selection.
-     * @param option The option component to search for.
-     * @param fireEvents Whether to fire change events.
-     */
-    private _selectOption(option: OptionComponent, fireEvents: boolean = true): OptionComponent | undefined {
-        if (!this._isOptionActive(option)) {
-            if (this._selected) {
-                this._selected.setSelected(false, false);
-            }
-            option.setSelected(true, false);
-            this._selected = option;
-            this._updateValue(fireEvents);
+        if (controlChange) {
+            this._updateValue(option.value);
             this.close();
-            return option;
         }
-        this.close();
-        return;
+        this._changeDetectorRef.detectChanges();
```

Comment:
You want to be careful when using detectChanges vs MarkForCheck. In this situation you are better with markForCheck. Think of situation where on the same level as this component you will for example some error logic to be displayed , so you want mark this  component and parent to be check in the next cycle so both are updated. 


Such excessive use of DetectChanges just make angular busy more than its needed. When you have a page with 50 similar controls and each of them runs detectChanges it will be pretty noisy, instead of marking component once and Angular can nicely update all marked to be checked. And you do use `OnPush`